test/built-ins/Iterator/from/get-return-method-when-call-return.js

```diff
@@ -0,0 +1,29 @@
+// Copyright (C) 2024 Sosuke Suzuki. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-iterator.from
+description: >
+  Gets the base iterator return method when the wrapper return method is called.
+info: |
+  %WrapForValidIteratorPrototype%.return ( )
+    ...
+    5. Let returnMethod be ? GetMethod(iterator, "return").
+
+features: [iterator-helpers]
+flags: []
+---*/
+
+let returnGetCount = 0;
+const iter = {
+    get return() {
+        returnGetCount++;
+        return function () {
+            return { value: 5, done: true };
+        };
+    },
+};
```

Comment:
This is totally optional and I'm fine with the test as written; but you could consider using the observer stuff from Temporal here, which would flag if the implementation erroneously used things like HasProperty or GetOwnPropertyDescriptor in addition to Get. I've been planning to move that into its own helper file that isn't named for Temporal. (If you'd prefer not to do that until after it's moved, that's fine too.)

```suggestion
const calls = [];
const iter = TemporalHelpers.propertyBagObserver(calls, {
  return() {
    return { value: 5, done: true };
  },
}, "originalIter");
```

You'd then assert things like
```js
assert.compareArray(calls, [
  "get originalIter.return",
]);
```

In fact  you could combine it with the following test, with something like
```js
const originalIter = {
  return() {
    return { value: 5, done: true };
  },
};
TemporalHelpers.observeMethod(calls, originalIter, "return", "originalIter");
const iter = TemporalHelpers.propertyBagObserver(calls, originalIter, "originalIter");

...

assert.compareArray(calls, [
  "get originalIter.return",
  "call originalIter.return",
]);
```
```